src/ReadLoader.cpp

```diff
@@ -507,7 +527,12 @@ void ReadLoader::readFile()
     int flags = O_RDONLY;
 #ifdef __linux__
     if(noCache) {
-        flags |= O_DIRECT;
+        flags = tryDirectIO(fileName);
+        if (flags == (O_RDONLY | O_DIRECT)) {
+            cout << "O_DIRECT flag supported by the file system." << endl;
+        } else {
+            cout << "O_DIRECT flag not supported by file system." << endl;
```

Comment:
I would rephrase in terms closer to the user. Something like "--Reads.noCache was turned off for file ... because it is not supported by the filesystem".